src/Temporalio/Client/Configuration/ClientConfig.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+
+namespace Temporalio.Client.Configuration
+{
+    /// <summary>
+    /// Represents the overall client configuration.
+    /// </summary>
+    public sealed class ClientConfig : ICloneable
```

Comment:
For many languages, this is generic "client config" name is fine because the namespace is often qualifying it at the callsite. But for .NET, I would recommend calling this `ClientEnvConfig`. Also, same for `ClientEnvConfigProfile` and `ClientEnvConfigTls`, though I wonder if we should consider just making those two, `Profile` and `Tls`, as inner classes here. Not sure, thoughts?